cmd/pw_ws_broker/web.go

```diff
@@ -217,24 +226,76 @@ func (bw *PwWsBrokerWeb) jsonGrid(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Content-Length", strconv.Itoa(len(gridJsonPayload)))
 		_, _ = w.Write(gridJsonPayload)
 	}
+}
+
+// search handles
+func (cl *ClientList) performSearch(query string) ws_protocol.SearchResult {
+
+	results := ws_protocol.SearchResult{
+		Aircraft: []*export.PlaneLocation{},
+		Airport:  []ws_protocol.AirportLocation{},
+		Route:    []string{},
+	}
+
+	if len(query) >= 3 {
+		// find any aircraft
+		cl.globalList.Range(func(key, value interface{}) bool {
+			loc := value.(*export.PlaneLocation)
+
+			if strings.Contains(strings.ToLower(loc.Icao), query) ||
```

Comment:
What's the performance of this like with a big data structure? Worth considering a better searching method?

Reply:
I think it's okay for now, even iterating through a few thousand items should be quick enough

and it's only used when people search (which probably won't be every second)